docs/getting_started/basic_usage/network.rst

```diff
@@ -0,0 +1,143 @@
+.. _network:
+
+=================
+Network Structure
+=================
+
+Construction
+------------------------------
+
+The network structure which defines the model topology is defined by the config ``network`` option,
+which is a dict, where each entry is a layer specification, which itself is a dict containing
+the kwargs for the specific layer class. E.g.:
+
+.. code-block:: python
+
+    network = {
+        "fw1": {"class": "linear", "activation": "relu", "dropout": 0.1, "n_out": 500},
+        "fw2": {"class": "linear", "activation": "relu", "dropout": 0.1, "n_out": 500, "from": ["fw1"]},
+        "output": {"class": "softmax", "loss": "ce", "from": ["fw2"]}
+    }
+
+The ``"class"`` key will get extracted from the layer arguments and the specific layer class will be used.
+For Theano, the base layer class is :py:class:`NetworkBaseLayer.Container` and :py:class:`NetworkBaseLayer.Layer`;
+for TensorFlow, it is :py:class:`TFNetworkLayer.LayerBase`.
+E.g. that would use the :py:class:`TFNetworkLayer.LinearLayer` class,
+and the ``LinearLayer.__init__`` will accepts arguments like ``activation``.
+In the given example, all the remaining arguments will get handled by the base layer.
+
+The construction itself can be found for TensorFlow in :py:func:`TFNetwork.TFNetwork.construct_from_dict`,
```

Comment:
Now you mix the usage and the technical implementation here. I'm not sure if that is a good idea.

Reply:
Btw, technological overview was supposed to give an overview over the technology, i.e. an overview over the implementation. There is certainly an overlap of such a technological overview and a usage overview, but it's not the same thing. I would say that an overview of the implementation is also useful to better understand how to use it, although it's not strictly necessary. It's similar of e.g. knowing how Git works (the object model etc) and how to use the tool.

I still prefer to keep those separate. Although it's totally ok if there is some overlap and some parts are redundant. E.g. to describe the implementation, it is certainly useful to do that given a simple network example.

Reply:
I agree that this needs some reworking, also by changing/extending the "technological overview" part. The question is if this rework should still be part of this pull request, or if it can be included in the upcoming documentation updates.

Reply:
The technological overview was supposed to give an overview of how it works for newcomers. Really only an overview, not going too much into detail.